### ISSUE TITLE
Simplify Chipmunk Code

### ISSUE THREAD

**Issue by smiths** [JacquesCarette/literate-scientific-software#247]
@hmadej, when you are between jobs, you might want to revise the existing simplified Chipmunk code.  When we spoke the other day I mentioned that it would be nice to have a simple implementation of the 2d game physics.  This would provide us with a target for our code generation.  I forgot at the time that Luthfi implemented something last summer.  You can find it in /CaseStudies/gamephys/code.  Luthfi got a subset of the original chipmunk to work for the simulation of a simple physics simulation.  It isn't for game play anymore, just for setting up a scene and running it.

I had a quick look at the code and it still seems more complex that what we would target with code generation.  Can you have a look at the code to see what we could simplify?  As you refactor the code, you can test it be making sure that the test cases Luthfi wrote continue to work.  You might also want to add additional test cases.

**Comment by hmadej** [comment]
do you want test cases for all the modules as well? right now I think there is only test cases for vector and bounding box.

**Comment by smiths** [comment]
Yes, we eventually want to test all of the modules.  I thought there was a test for an object falling under its own weight, but I might be mis-remembering.

**Comment by smiths** [comment]
Yes, we can close this issue.  Someday we might want a manual implementation of the game physics engine, since Chipmunk is more complicated than we want as a starting point.  This isn't a high priority though.  We don't even know for sure what direction we'll go in for the code generation for the game physics engine.  It makes more sense to get some of the other examples working first.
